prefect_aws/secrets_manager.py

```diff
@@ -350,3 +352,152 @@ def example_delete_secret_with_recovery_window():
     except ClientError:
         logger.exception("Unable to delete secret %s", secret_name)
         raise
+
+
+class SecretsManager(SecretBlock):
+    """
+    Block that represents a resource that can store and retrieve secrets.
+
+    Attributes:
+        aws_credentials: The credentials to use for authentication with AWS.
+        secret_name: The name of the secret.
+    """
+
+    aws_credentials: Union[AwsCredentials, MinIOCredentials]
```

Comment:
Related to the comment above about MinIO, this should only accept `AwsCredentials`.
```suggestion
    aws_credentials: AwsCredentials
```